content/dakshdeephere/threading.md

```diff
@@ -0,0 +1,355 @@
+---
+Title: "Introduction to threading in C"
+Description: "Step-by-step explain about what are threads and how threading works with suitable code samples."
+DatePublished: "2023-06-04"
+Categories:
+  - "operating-system"
+  - "python"
+  - "concurrency"
+  - "computer-science"
+Tags:
+  - "Algorithm"
+  - "Node"
+CatalogContent:
+  - "learn-python"
+  - "paths/threading"
+---
+
+_**Prerequisites:** Understanding of C, Operating System, Visual studio code_  
+_**Version:** Standard C language versions C89/C90, C99, C11, and C18_
+
+## Introduction
+
+A `thread` is a path of execution within a process, which can contain multiple threads. It is also known as a lightweight process. The goal of threading is to achieve parallelism by dividing a process
+into multiple threads. For example, in a browser, multiple tabs can be different threads. MS Word uses multiple threads: one to format the text and another to process inputs, etc.
+
+**`Thread`** is an execution unit that consists of its own program `counter`, a `stack`, and a set of `registers`. `Program counter` keeps track of which instruction to execute next, the system registers which hold its current working variables, and a stack that contains the execution history.
+
+**A thread comprises of:**
```

Comment:
Would work well as a numbered list

Reply:
It'd also be good to have a short explanation on what each is

Reply:
> Would work well as a numbered list

Haha I'm so used to this listing that I've literally stopped using the numbered list

Reply:
> It'd also be good to have a short explanation on what each is

Got it